gk/main.c

```diff
@@ -386,33 +416,72 @@ setup_gk_instance(unsigned int lcore_id, struct gk_config *gk_conf)
 	return ret;
 }
 
-static void
-add_ggu_policy(struct ggu_policy *policy, struct gk_instance *instance)
+static struct flow_entry *
+add_new_flow_from_policy(struct ggu_policy *policy,
+	struct gk_instance *instance, struct gk_lpm *ltbl)
 {
 	int ret;
-	uint64_t now = rte_rdtsc();
-	struct flow_entry *fe;
+	int gt_id = -1;
+	struct flow_entry *fe = NULL;
 	uint32_t rss_hash_val = rss_ip_flow_hf(&policy->flow, 0, 0);
 
+	/*
+	 * When the flow entry already exists,
+	 * the grantor ID should be already known.
+	 * Otherwise, Grantor ID comes from LPM lookup.
+	 */
 	ret = rte_hash_lookup_with_hash(instance->ip_flow_hash_table,
```

Comment:
What's this lookup doing here? This function is to **ADD** a new flow. If `add_new_flow_from_policy()` is called, the caller already knows it's a **NEW** flow.